lib/store/src/csf/stepRunners.ts

```diff
@@ -0,0 +1,13 @@
+import { AnyFramework, StepRunner } from '@storybook/csf';
+
+export function composeStepRunners<TFramework extends AnyFramework>(
+  stepRunners: StepRunner<TFramework>[]
+): StepRunner<TFramework> {
+  return async (label, play, playContext) => {
+    const composedPlay = await stepRunners.reduce(
+      async (innerPlay, stepRunner) => async () => stepRunner(label, await innerPlay, playContext),
+      Promise.resolve(play)
+    );
+    await composedPlay(playContext);
+  };
+}
```

Comment:
This function needs a big explainer. The code is very dense and if you don't know why this exists then you'll have a hard time understanding what's going on.